parsec/crypto.py

```diff
@@ -81,14 +81,40 @@ def hmac(self, data: bytes, digest_size=BLAKE2B_BYTES) -> bytes:
         return blake2b(data, digest_size=digest_size, key=self, encoder=RawEncoder)
 
 
-class HashDigest(bytes):
-    __slots__ = ()
+class HashDigest:
+    __slots__ = ("_digest",)
+
+    def __init__(self, digest: bytes):
+        self._digest = digest
+
+    def __repr__(self) -> str:
+        return f"HashDigest({self.hexdigest()})"
+
+    def __eq__(self, other):
+        if not isinstance(other, HashDigest):
+            return NotImplemented
+        return self.digest == other.digest
 
     @classmethod
     def from_data(self, data: bytes) -> "HashDigest":
         # nacl's sha256 doesn't accept bytearray, so stick to `hashlib.sha256`
         return HashDigest(sha256(data).digest())
 
+    @property
+    def digest(self) -> bytes:
+        return self._digest
+
+    def hexdigest(self) -> str:
+        return self._digest.hex()
+
+
+if not TYPE_CHECKING:
+    # Try to overwrite Python implementation with Rust one
+    try:
+        from libparsec.hazmat import HashDigest  # noqa: F811
+    except ImportError:
+        pass
```

Comment:
Maybe expose `_PyHashDigest` and `_RstHashDigest` so they can be imported from tests.

This way it's also possible to do `assert _RstHashDigest is HashDigest` to check that the rust extension is available (and not silently ignored).

Reply:
Good idea !